### ISSUE TITLE
Invariant Violation: new NativeEventEmitter() requires a non-null argument

### ISSUE THREAD

**Issue by SalmanCorleone** [inbrainai/sdk-react#68]
Invariant Violation: new NativeEventEmitter() requires a non-null argument
-
Facing error right after installing 'inbrain-surveys' for react-native app. My assumption is the following line is causing this issue:
```
const inbrainEmitter = new NativeEventEmitter(InBrainSurveys);
```

Version: 2.13.0
Build platform: iOS
React native version: 0.71.14

Thanks in advance

**Comment by susslik** [comment]
Hi @SalmanCorleone, thanks for reporting this! We will take a look and back to you shortly

**Comment by SalmanCorleone** [comment]
React native version: 0.71.14 @susslik

**Comment by susslik** [comment]
Thanks @SalmanCorleone! Could you pls try the following steps:
Go to ios folder then:
1) Remove Pods & Podfile.lock
2) open terminal and run "pod install"
3) Rebuild project

Please, let me know if that helps

**Comment by beqramo** [comment]
Hi @susslik,
the problem is exactly on this line:
https://github.com/inbrainai/sdk-react/blob/9a21a5be75954ab9c7b2526d0614e8ac305b80f9/src/InBrain.tsx#L15

The native module isn't getting attached to JS nativeModule so it is null and NativeEventEmitter can't have null.

On android, it works perfectly.

Something is wrong with the package configuration on the native iOS side.

same version.

**Comment by susslik** [comment]
Hi @beqramo, thanks for letting us know!
Could you pls try the following steps:
Go to ios folder then:

1) Remove Pods & Podfile.lock;
2) Open terminal and run "pod install";
3) Ensure that pods installed succeeded and no errors were raised;
4) Rebuild project.

Please, let me know if that helps

**Comment by beqramo** [comment]
No, it doesn't as pods are getting installed without any problem.
Also if it were a pod error it wouldn't even build because there would be a linking problem.
So it is getting linked well just nativemodules aren't assigned well on native side.

**Comment by susslik** [comment]
Got it, thanks for the details @beqramo!
Digging into the issue, that seems pretty interesting! Will keep you posted

**Comment by susslik** [comment]
Hi @beqramo, @SalmanCorleone - hope you're doing well! 
We were digging into the trouble and managed to reproduce it - please find our experience bellow:
1) Create the new app using RN 0.71.4;
2) Install pods;
3) Add  "inbrain-surveys": "^2.13.0" package and run _npm install_;
4) Try to run iOS app - and the build succeeded, but got the mentioned error after app launch;
5) Install pods;
5) Run iOS app again - everything works as expected!

That was the only way which led us to the mentioned error, no other options to reproduce it were found. Based on that we tend to some trouble with the pods and would like to propose the following steps:
- clean all the caches (pods, RN, remove the app from the simulator/device, clean derived data, etc);
- install the pods.

Please, let us know if you have some other way to reproduce the issue so we can continue our investigation!

**Comment by beqramo** [comment]
Okay, I will do it, but as a fact that you faced it the first time. think it already means that something isn't correct.

Also, it isn't a stable solution as on CI/CD pipeline I can't expect some build errors or launch problems.

I would suggest checking the problem, still.

Thanks, I will update you.

**Comment by susslik** [comment]
Hi @beqramo - thanks, will be waiting for the update! In the meantime let me describe that case a bit to highlight some additional details:
We have faced with that first time as the package was added, but the _pod install_ has not been done. And that's expected behaviour, as _npm install_ is adding only RN package, without it's native dependencies.
Taking into account that the package relies on native modules and in order to make it work at iOS - those modules have to be installed as well. And that's why the _pod install_ command is required after the package was added and _npm install_ completed.
That's a common flow for the RN modules with native submodules, for ex. [Crashlytics](https://rnfirebase.io/crashlytics/usage#installation), [Lottie](https://github.com/lottie-react-native/lottie-react-native) and so on...

Based on that - I'd say that this is not a problem, that's the expected flow for the packages with native dependencies. Our bad that we have not stated this step at our readme, will fix that ASAP!

**Comment by beqramo** [comment]
Oh, I thought you you ran it, my mistake, I assumed at first it didn't work.

Anyway, I tried the suggested and it didn't work.

So I took some time and I found that this line is the problematic, removing it, fixes it:
https://github.com/inbrainai/sdk-react/blob/9a21a5be75954ab9c7b2526d0614e8ac305b80f9/inbrain-surveys.podspec#L27
I don't think there is any need to have that line anymore.


Also I took a look at your native codes and RCTEventEmitter implementation doesn't follow the pattern. if you will have time please take a look later.

Thanks

**Comment by susslik** [comment]
Thanks for the update @beqramo, we will take a look!
Very interesting why that line causes the trouble... Could you please let me know the additional details so we can dig a bit more:
- which processor your mac has (intel/m1/m2/m3)?
- which Xcode version you're using?
- which device or simulator you're using for testing?
- which version of Cocoapods installed at your mac?

Want to find out what exactly led to the issue, as we have tried a few different laptops and have no trouble with that line...

**Comment by beqramo** [comment]
It happened because i have m1 mac.
I'm using Xcode 15 and ios 17 for testing. pod version is 1.13.0

**Comment by beqramo** [comment]
Yes it is a simulator.

Check other libraries, they don't have it too... XCode 12 is a long gone so doesn't make sense to have that line at all.

**Comment by susslik** [comment]
Yeah, however looks like the issue also exists in newer versions of Xcode, for ex. Xcode 14 and Intel-based Macs.
In general - I agree that this line looks outdated, however looks like we have to remove it at the native module's .podspec as well. 
Expect to release the new versions of both native and RN modules early next week, most probably on Monday.

Thank you for your patience!

**Comment by susslik** [comment]
Hi @beqramo, @SalmanCorleone - we have released the new SDK version 2.15.0, could you please try it and let us know if the issue has been solved
